app/src/main/java/com/hedvig/app/service/push/PushNotificationService.kt

```diff
@@ -55,53 +45,21 @@ class PushNotificationService : FirebaseMessagingService() {
     }
 
     override fun onMessageReceived(remoteMessage: RemoteMessage) {
-        when (remoteMessage.data[NOTIFICATION_TYPE_KEY]) {
-            NOTIFICATION_TYPE_NEW_MESSAGE ->
-                ChatNotificationManager
-                    .sendChatNotification(this, remoteMessage)
-            NOTIFICATION_TYPE_REFERRAL_SUCCESS ->
-                ReferralsNotificationManager
-                    .sendReferralNotification(this, remoteMessage)
-            NOTIFICATION_TYPE_REFERRALS_ENABLED ->
-                ReferralsNotificationManager
-                    .sendReferralsEnabledNotification(this)
-            NOTIFICATION_TYPE_CONNECT_DIRECT_DEBIT ->
-                paymentNotificationManager
-                    .sendDirectDebitNotification(this)
-            NOTIFICATION_TYPE_PAYMENT_FAILED ->
-                paymentNotificationManager
-                    .sendPaymentFailedNotification(this)
-            NOTIFICATION_TYPE_GENERIC_COMMUNICATION ->
-                GenericNotificationManager
-                    .sendGenericNotification(this, remoteMessage)
-            NOTIFICATION_CROSS_SELL ->
-                crossSellNotificationManager
-                    .sendCrossSellNotification(this, remoteMessage)
-            NOTIFICATION_TYPE_REFERRALS_CAMPAIGN ->
-                ReferralsNotificationManager
-                    .sendReferralCampaignNotification(this, remoteMessage)
-            else ->
-                ChatNotificationManager
-                    .sendDefaultNotification(this, remoteMessage)
+        notificationSenders.forEach { sender ->
+            if (sender.handlesNotificationType(remoteMessage.data[NOTIFICATION_TYPE_KEY])) {
```

Comment:
I decided to go for the loop short-circuiting here, as you never want more than one notification to be shown from one notification-event (as far as I could think of, at least)

Reply:
Yeah makes total sense. Should it maybe be a chain of .firstNotNullOfOrNull()?.sendNotification instead to make the short-circuiting feature of this a bit more explicit?

Something like (wrote outside of IDE):
notificationSenders.firstNotNullOfOrNull { sender ->
  sender.handlesNotificationType(remoteMessage.data[NOTIFICATION_TYPE_KEY])
}.sendNotification(remoteMessage)